### ISSUE TITLE
When training the TensorFlow Lite model after loading it in C++, the logit/probabllites size is different from the number of y labels.

### ISSUE THREAD

**Issue by ysbaekFox** [tensorflow/tensorflow#58070]
### Issue Type

Bug

### Source

source

### Tensorflow Version

tf 2.9.1

### Custom Code

No

### Python version

3.9

### Current Behaviour?

```shell
When I ran the python model, as a result of predict, the size of logit and probablites was the same as the size of y label.  
But, in C++, 
The size of logits is 0, and the size of proba is 6. Am I using C++ Code wrong?
```

### Standalone code to reproduce the issue

```shell
import pickle
import tensorflow as tf
from tensorflow.keras.layers import Dense, Activation
from tensorflow.keras.models import Model
from tensorflow.keras.utils import get_custom_objects

tf.keras.backend.clear_session()

class Relu(Activation):
    def __init__(self, activation, **kwargs):
        super(Relu, self).__init__(activation, **kwargs)
        self.__name__ = 'Relu'

def _relu(tensor):
    return tf.maximum(tensor,0)

get_custom_objects().update({'_relu': Relu(_relu)})

class PredictedDestination(tf.Module):
    def __init__(self):
        self.model = tf.keras.Sequential([
            tf.keras.layers.Dense(16, input_shape=(18, ), name='input'),
            tf.keras.layers.Dense(5),
        ])
    
        self.model.compile(
            optimizer='sgd',
            loss=tf.keras.losses.CategoricalCrossentropy(from_logits=True),
            run_eagerly=True)

    # The `train` function takes a batch of input images and labels.
    @tf.function(input_signature=[
      tf.TensorSpec([None, 18], tf.float32),
      tf.TensorSpec([None, 5], tf.float32),
    ])
    def train(self, x, y):
        epochs = 100
        for i in range(epochs):
            with tf.GradientTape() as tape:
              prediction = self.model(x)
              loss = self.model.loss(y, prediction)
            gradients = tape.gradient(loss, self.model.trainable_variables)
            self.model.optimizer.apply_gradients(zip(gradients, self.model.trainable_variables))
            result = {"loss": loss}
        return result

    @tf.function(input_signature=[
      tf.TensorSpec([None, 18], tf.float32),
    ])
    def predict(self, x): 
        logits = self.model(x)
        probabilities = tf.nn.softmax(logits, axis=-1)
        print(probabilities)
        print(logits)
        return {
            "output": probabilities,
            "logits": logits
        }

model = PredictedDestination()



```python
SAVED_MODEL_DIR = "predicted_destination_model"

tf.saved_model.save(
    model,
    SAVED_MODEL_DIR,
    signatures={
        'train':
            model.train.get_concrete_function(),
        'predict':
            model.predict.get_concrete_function(),
    })

converter = tf.lite.TFLiteConverter.from_saved_model(SAVED_MODEL_DIR)
converter.target_spec.supported_ops = [
    tf.lite.OpsSet.TFLITE_BUILTINS,
    tf.lite.OpsSet.SELECT_TF_OPS
]
converter.experimental_enable_resource_variables = True
tflite_model = converter.convert()

f = open('predicted_destination.tflite', 'wb')
f.write(tflite_model)
f.close()
```

```c++
tflite::SignatureRunner* trainSignatureRunner = m_interpreter->GetSignatureRunner( "train" );

trainSignatureRunner->ResizeInputTensor( "x", { 1, 18 } );
trainSignatureRunner->ResizeInputTensor( "y", { 1, 5 } );

trainSignatureRunner->AllocateTensors()

TfLiteTensor* x_input_tensor = trainSignatureRunner->input_tensor( "x" );
// fill x_input_tensor

TfLiteTensor* y_input_tensor = trainSignatureRunner->input_tensor( "y" );
// fill y_input_tensor

trainSignatureRunner->Invoke();

tflite::SignatureRunner* predictSignatureRunner = m_interpreter->GetSignatureRunner("predict");

predictSignatureRunner->ResizeInputTensor( "x", { 1, 18 } );
TfLiteTensor* predict_x_input_tensor = predictSignatureRunner->input_tensor( "x" );
// fill predict_x_input_tensor

predictSignatureRunner->Invoke();

std::vector<FLOAT> logitsResult;
FLOAT* tempLogits = logit_output->data.f;

while ( 0 < *tempLogits )
{
	std::cout << "[TFLiteModelManager::Predict] logit : " << *tempLogits << std::endl;
	logitsResult.push_back( *tempLogits );
	tempLogits++;
}

const TfLiteTensor* probabilities_output = predictSignatureRunner->output_tensor( "output" );
std::vector<FLOAT> probabilities;
FLOAT* tempProbabilities = probabilities_output->data.f;

while ( 0 < *tempProbabilities )
{
	std::cout << "[TFLiteModelManager::Predict] proba : " << *tempProbabilities << std::endl;
	probabilities.push_back( *tempProbabilities );
	tempProbabilities++;
}
```
```


### Relevant log output

```shell
[TFLiteModelManager::Predict] proba : 0.112619
[TFLiteModelManager::Predict] proba : 0.170277
[TFLiteModelManager::Predict] proba : 0.224098
[TFLiteModelManager::Predict] proba : 0.301938
[TFLiteModelManager::Predict] proba : 0.191067
[TFLiteModelManager::Predict] proba : 0.252747

logits size is zero, but proba size size is 6....
```

**Comment by ysbaekFox** [comment]
In python code, If two types are returned as shown below, a problem seems to occur.
I removed the code that returned "logits", left only "output", and it worked normally.

```
...
...
  return {
      "output": probabilities,
      "logits": logits
  }
```

**Comment by pkgoogle** [comment]
Hi @ysbaekFox, can you either show your full code or a compilable toy example which shows this issue?

I'm trying to answer what batchSize is and what FLOAT actually is, my guess is it's a typedef but assuming these things usually lead to trouble.
